### PR TITLE
Second ELMS iteration

### PR DESCRIPTION
**Pull request by MagdaPaj**
## Purpose
* add iothub, iotedgedevice, iotedgevm modules
* create 2 ELMS environments: basic-elms (as we had in the first iteration) and extended-elms (full environment with new IoT Hub, Edge device, VM)
* update documentation
* small improvements to the ELMS module

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
* follow the documentation from the provision-elms-with-terraform.md

### CODE REVIEW

File: terraform/environments/extended-elms/output.tf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+output "vm_userpassword" {
+  value = module.iotedgevm.vm_userpassword
```
Comment (MagdaPaj) [inline terraform/environments/extended-elms/output.tf]: What do we do with this? It's probably not the best idea to output it from the security perspective. So options that I see:
1. add Key Vault for storing it. But we don't have key vault in PowerShell script. And we will store there only this.
2. Do not output the VM password, but force user to provide it as a variable.

File: terraform/modules/iotedgevm/main.tf [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  allocation_method   = "Dynamic"
+  domain_name_label   = "${local.dns_label_prefix}-${var.random_id}"
+}
+
+resource "azurerm_network_security_group" "iot_edge" {
+  name                = "${local.dns_label_prefix}-nsg"
+  resource_group_name = var.rg_name
+  location            = var.location
+
+  security_rule {
```
Comment (Ioana37) [inline terraform/modules/iotedgevm/main.tf]: Should we leave SSH port open by default?

**Review by Ioana37 — APPROVED** [review]

File: terraform/modules/iotedgevm/main.tf (hunk 2 above)
Comment (MagdaPaj) [reply terraform/modules/iotedgevm/main.tf]: Good question. But I don't know. If we delete it, then a user won't be able to access the VM with the current setup. But not sure if that's really needed in the ELMS. Probably not. And it's more secure without it.

File: terraform/environments/extended-elms/output.tf (hunk 1 above)
Comment (MagdaPaj) [reply terraform/environments/extended-elms/output.tf]: Just a note, that the PS deploy script also outputs it.

Comment (Ioana37) [reply terraform/environments/extended-elms/output.tf]: We agreed to not output the VM password.

File: terraform/modules/iotedgevm/main.tf (hunk 2 above)
Comment (Ioana37) [reply terraform/modules/iotedgevm/main.tf]: We decided to not leave SSH port open. If needed user can find documentation on how to access a VM.

**Pull request merged**